client/components/spinner/index.jsx

```diff
@@ -8,140 +8,30 @@ import PropTypes from 'prop-types';
-			// in componentDidMount()
-			isSVGCSSAnimationSupported: false,
-		};
-	}
-
-	componentWillMount() {
-		this.setState( {
-			instanceId: ++Spinner.instances,
-		} );
-	}
-
-	componentDidMount() {
-		if ( isSVGCSSAnimationSupported ) {
-			// Turning off eslint rule on this line as an exception — we want to trigger
-			// a re-render if we're progressively enhancing with SVG animations.
-			// eslint-disable-next-line react/no-did-mount-set-state
-			this.setState( {
-				isSVGCSSAnimationSupported: isSVGCSSAnimationSupported,
-			} );
-		}
-	}
-
-	getClassName() {
-		return classNames( 'spinner', this.props.className, {
-			'is-fallback': ! this.state.isSVGCSSAnimationSupported,
-		} );
-	}
+	render() {
+		const className = classNames( 'spinner', this.props.className );
 
-	renderFallback() {
 		const style = {
 			width: this.props.size,
 			height: this.props.size,
+			fontSize: this.props.size, // allows border-width to be specified in em units
 		};
 
 		return (
-			<div className={ this.getClassName() } style={ style }>
-				<span className="spinner__progress is-left" />
-				<span className="spinner__progress is-right" />
-			</div>
-		);
-	}
-
-	render() {
-		if ( ! this.state.isSVGCSSAnimationSupported ) {
-			return this.renderFallback();
-		}
-
-		const { instanceId } = this.state;
-
-		return (
-			<div className={ this.getClassName() }>
-				<svg
-					className="spinner__image"
-					width={ this.props.size }
-					height={ this.props.size }
-					viewBox="0 0 100 100"
-				>
-					<defs>
-						<mask id={ `maskBorder${ instanceId }` }>
-							<rect x="0" y="0" width="100%" height="100%" fill="white" />
-							<circle r="46%" cx="50%" cy="50%" fill="black" />
-						</mask>
-						<mask id={ `maskDonut${ instanceId }` }>
-							<rect x="0" y="0" width="100%" height="100%" fill="black" />
-							<circle r="46%" cx="50%" cy="50%" fill="white" />
-							<circle r="30%" cx="50%" cy="50%" fill="black" />
-						</mask>
-						<mask id={ `maskLeft${ instanceId }` }>
-							<rect x="0" y="0" width="50%" height="100%" fill="white" />
-						</mask>
-						<mask id={ `maskRight${ instanceId }` }>
-							<rect x="50%" y="0" width="50%" height="100%" fill="white" />
-						</mask>
-					</defs>
-					<circle
-						className="spinner__border"
-						r="50%"
-						cx="50%"
-						cy="50%"
-						mask={ `url( #maskBorder${ instanceId } )` }
-					/>
-					<g mask={ `url( #maskDonut${ instanceId } )` }>
-						<g mask={ `url( #maskLeft${ instanceId } )` }>
-							<rect className="spinner__progress is-left" x="0" y="0" width="50%" height="100%" />
-						</g>
-						<g mask={ `url( #maskRight${ instanceId } )` }>
-							<rect
-								className="spinner__progress is-right"
-								x="50%"
-								y="0"
-								width="50%"
-								height="100%"
-							/>
-						</g>
-					</g>
-				</svg>
+			<div className={ className } style={ style }>
+				<div className="spinner__outer">
+					<div className="spinner__inner" />
+				</div>
```

Comment:
Any reason for creating two divs instead of the `:after` approach I took with the CodePen?

Reply:
The spinner styles (mainly colors) are customized at many places where it's used. I just thought that having real class names for them instead of pseudo-classes is a better "developer experience".

There is no problem at all with the `::before` styles per se.

I expect to do a few more tweaks when I go over all the `.spinner` custom styles.

Reply:
I think an extra div is clearer in this case.